components/script/script_thread.rs

```diff
@@ -2114,30 +2114,29 @@ impl ScriptThread {
             wrapper: None,
         };
         ROUTER.add_route(action_receiver.to_opaque(), box move |message| {
-            listener.notify_action(message.to().unwrap());
+            listener.notify_fetch(message.to().unwrap());
         });
-        let response_target = AsyncResponseTarget {
-            sender: action_sender,
-        };
 
         if load_data.url.scheme() == "javascript" {
             load_data.url = Url::parse("about:blank").unwrap();
         }
 
-        self.resource_threads.send(CoreResourceMsg::Load(NetLoadData {
-            context: LoadContext::Browsing,
-            url: load_data.url,
+        let request = RequestInit {
+            url: load_data.url.clone(),
             method: load_data.method,
-            headers: Headers::new(),
-            preserved_headers: load_data.headers,
-            data: load_data.data,
-            cors: None,
+            destination: Destination::Document,
+            credentials_mode: CredentialsMode::Include,
+            use_url_credentials: true,
+            origin: load_data.url,
             pipeline_id: Some(id),
-            credentials_flag: true,
+            referrer_url: load_data.referrer_url,
             referrer_policy: load_data.referrer_policy,
-            referrer_url: load_data.referrer_url
-        }, LoadConsumer::Listener(response_target), None)).unwrap();
+            headers: load_data.headers,
+            body: load_data.data,
+            .. RequestInit::default()
```

Comment:
Let's file a followup issue to support non-follow redirect modes for RequestInit.